### DISCUSSION TITLE
Compatibility with `package.json` files?

### DISCUSSION THREAD

**Discussion by balupton** [nestdotland/nest.land#239] [General]
I've been using [make-deno-edition](https://github.com/bevry/make-deno-edition) to provide deno compatibility now for 30+ of the bevry packages, which were originally targetting node or web browsers.

Reading https://docs.nest.land/eggs/configuration.html#field-information it seems nest needs an eggs.json file, and when I click the link to find out why it doesn't use package.json, it took me to https://github.com/nestdotland/nest.land/issues/52#issuecomment-643038042 which returns the error:

> This issue has been moved to a repository you don't have access to or has been deleted since being transferred.

Tooling like `make-deno-edition` and `projectz` utilise the npm registry, and use unpkg for the deno cdn. I would like to make it so that `make-deno-edition` also provides publishing to the various deno registries until there is a clear winner.

However, it would be a lot easier for me (the author of `make-deno-edition`) if nest just used package.json instead, and did any custom configuration within a `nest` field inside it.

**Comment by t8** [comment]
Hi @balupton! Thanks so much for letting us know that the package.json link was broken. We'll get that fixed! 

I understand your enthusiasm for `package.json`, however, our end goal is to be fully compatible with projects without needing any configuration file at all. `egg.json` will eventually become a fully optional file and all configuration will be able to be completed in alternative ways.

I think that what you're doing with Bevry and make-deno-edition is really interesting, and I really appreciate you taking interest in nest.land. With that said, I think that there could be other ways for you to introduce compatibility to the platform. As you know, our system is catered directly towards Deno users, which means that we don't support any sort of "start" commands like what `package.json` includes.

I'm not completely sure how make-deno-edition makes modules compatible with Deno, but I do know that you could figure out all of the required fields in `egg.json` like so:

```js
{
    "name": "module-name", // Equivalent to "name" in package.json
    "description": "Your brief module description", // Equivalent to "description" in package.json
    "version": "0.0.1", // Equivalent to "version" in package.json
    "entry": "./src/main.ts", // Not completely sure, but this is most likely possible with make-deno-edition
    "repository": "https://github.com/your_name/your_project", // Equivalent to "repository.url" in package.json
}
``` 

Note that I left out the `"files"` field. This is because you can use an `.eggignore` file instead. You could just create a universal template `.eggignore` file that automatically ignores all node files, the `.git` folder, and can even include all files in a user's `.gitignore`. This way, you wouldn't have to devise an algorithm to figure out which files to include.

<hr />
The .eggignore feature support is coming in our next release. You can expect this within the next 3 days!
<hr />

I should also mention that we're trying to be so platform-independent here because we don't want to restrict the user to using some sort of source control or external service for their projects. 

I'd like to hear your thoughts on this, so feel free to keep the conversation going.
